File: Python/IsPalindromePerm.py
def is_palindrome_perm(string):
    #print(string)
    counts = [False]*128
    for i in range(0, len(string)):
        index = ord(string[i])
        counts[index] = not counts[index]

    oddFound = False
    for i in range(0, 128):
        if counts[i] and oddFound:
            return False

        if counts[i]:
            oddFound = True

    return True

File: Java/CTCI/src/strings_and_arrays/IsPalindromePerm.java
package strings_and_arrays;

public class IsPalindromePerm {

	public static boolean is_palindrome_perm(String s) {

		boolean[] counts = new boolean[128];
		for (int i = 0; i < s.length(); i++) {
			int index = s.charAt(i);
			counts[index] = !counts[index];
		}
		boolean oddFound = false;
		for (int i = 0; i < 128; i++) {
			if (counts[i] && oddFound) {
				
				return false;
			}
			if (counts[i]) {
				oddFound = true;
			}
		}
		return true;
	}
}
